### ISSUE TITLE
Include ProviderIds in /sync/watched/

### ISSUE THREAD

**Issue by ginjaninja1** [trakt/api-help#363]
might it be possible to include the providerids (tvdb in particular) for episodes in the response. So that when the index on trakt does not match the index on the application it is still possible to map watched events to episodes with same providerids. The need is particularly prevalent on specials eg https://trakt.tv/shows/crazy-ex-girlfriend/seasons/0/episodes/2 does not agree with the index on tvdb (and my app) but does agree with the tvdbid.

thanks for considering
this is a feature request but havent worked out how to specify that (yet)

**Comment by rectifyer** [comment]
The various IDs are included in method such as https://trakt.docs.apiary.io/#reference/episodes/summary/get-a-single-episode-for-a-show and several other methods as well.

**Comment by tysonkerridge** [comment]
You may be using the wrong `type` in `/users/:id/history/:type` 
eg. `GET https://api.trakt.tv/users/me/history/episodes` should return all the auth user's episodes, paginated. These episode objects would include the `tvdb` ID.

**Comment by ginjaninja1** [comment]
> You may be using the wrong `type` in `/users/:id/history/:type` eg. `GET https://api.trakt.tv/users/me/history/episodes` should return all the auth user's episodes, paginated. These episode objects would include the `tvdb` ID.

thank you..thats working for me and providing the info as you say. but i dont think its efficient for a full sync of a users watched status on epsiodes...not least because every instance of a watched action is communicated.rather than a single watched for each index.

so undeterred i used what you showed about /users and tried `https://api.trakt.tv/users/me/watched/episodes`. This showed the episodes and ids but not the show the episode belonged to
`plays           : 8
last_watched_at : 27/07/2019 23:46:42
last_updated_at : 27/07/2019 23:46:42
episode         : @{season=1; number=14; title=Law and Disorder in Lagos; ids=}`

so i tried `https://api.trakt.tv/users/me/watched/shows` and that provides the show and the episode indexes but not the tvdbids.

so it maybe that i will have to iterate through each individual known show to get an efficient watched list with ids for a known show?  (if you havent guessed yet im very unfamiliar with trakt api so appreciate all the pointers i can get)

to a ley man it doesnt an unreasonable feature request to get the episode providerids added to the /watched endpoint[s]?

**Comment by tysonkerridge** [comment]
Yeah your suggestion of a way to get episode ids from the `/watched/shows` endpoint may be the way around those issues though still doesn't really solve the inefficiency either, I imagine.

**Comment by Zippy79** [comment]
Are there currently any plans to add episode ids to `/watched/shows`?
